src/components/Navbar/Navbar.cy.tsx

```diff
@@ -0,0 +1,89 @@
+import Navbar, { NavbarItem } from "./Navbar";
+import {
+  faClock,
+  faCompass,
+  faFutbol,
+  faUser,
+} from "@fortawesome/free-solid-svg-icons";
+
+const initialItems = [
+  {
+    icon: faClock,
+    title: "Time",
+  },
+  {
+    icon: faFutbol,
+    title: "Comments",
+  },
+  {
+    icon: faCompass,
+    title: "Compass",
+  },
+  {
+    icon: faUser,
+    title: "User",
+  },
+];
+
+describe("<Navbar />", () => {
+  it("renders navbar with 4 items", () => {
+    cy.mount(
+      <Navbar label="Navbar">
+        {initialItems.map((item, i) => (
+          <NavbarItem
+            key={`${item.title}-${i}`}
+            icon={item.icon}
+            iconStyles={{ fontSize: "30px" }}
+            title={item.title}
+            onSelect={() => {}}
+          />
+        ))}
+      </Navbar>
+    );
+  });
+
+  it("should set first item as default active", () => {
+    cy.mount(
+      <Navbar label="Navbar" defaultActive={0}>
+        {initialItems.map((item, i) => (
+          <NavbarItem
+            key={`${item.title}-${i}`}
+            icon={item.icon}
+            iconStyles={{ fontSize: "30px" }}
+            title={item.title}
+            onSelect={() => {}}
+          />
+        ))}
+      </Navbar>
+    );
+    cy.get(".spot .dot").should("exist");
+    cy.get('[data-testid="clock-icon"]').should("have.class", "active");
+  });
+
+  it("should fire an event and change active item on click", () => {
+    const onClickSpy = cy.spy().as("onClickSpy");
+
+    cy.mount(
+      <Navbar label="Navbar">
+        {initialItems.map((item, i) => (
+          <NavbarItem
+            key={`${item.title}-${i}`}
+            icon={item.icon}
+            iconStyles={{ fontSize: "30px" }}
+            title={item.title}
+            onSelect={onClickSpy}
+          />
+        ))}
+      </Navbar>
+    );
+
+    cy.get(".spot .dot").should("not.exist");
+    cy.get('[data-testid="clock-icon"]').should("not.have.class", "active");
+
+    cy.get(".navbar-item-btn").first().click();
+
+    cy.get("@onClickSpy").should("have.been.called");
+    cy.get(".spot .dot").should("exist");
+    cy.get('[data-testid="clock-icon"]').should("have.class", "active");
```

Comment:
suggestion: to go a step even further, you could test that when clicking another item (eg the 3rd one), the dot moves to the expected position. (assert the pixels here).

P.S. This test is considered a "visual test" which, normally is done with tools like [Percy](https://percy.io/), if you wanna give it a shot. (you can integrate it with Cypress!)

Reply:
added Percy 💪, it was weird at first since I was expecting to assert things using code that looks more like Cypress, but Percy just relays on the snapshots u take (at least from what I understood) and compares them with the images that are uploaded

Reply:
Yeah, exactly. Percy is just visual comparison, nothing more. Maybe with ChatGPT it will be more powerful to describe the differences eheh